chain/transaction.go

```diff
@@ -377,6 +379,81 @@ func (t *Transaction) Marshal(p *codec.Packer) error {
 	return t.marshal(p)
 }
 
+type txJSON struct {
+	ID      ids.ID      `json:"id"`
+	Actions codec.Bytes `json:"actions"`
+	Auth    codec.Bytes `json:"auth"`
+	Base    codec.Bytes `json:"base"`
```

Comment:
Could we keep `Base` as the original type since we should be able to encode/decode this directly as JSON with no issues?